### PR TITLE
Adding comments/documentation for APIs

### PR DESCRIPTION
**Pull request by vivekramamoorthy**

### CODE REVIEW

File: lib/topology_lib_tcpdump/library.py [hunk 1]
```diff
@@ -29,13 +29,19 @@
...
 # Add your library functions here.
 
-def tcpdump_rate(sw):
+"""
+Get packet rate in packets per second after capture has been run.
+:param enode: Engine node to communicate with.
+:type enode: topology.platforms.base.BaseNode
+"""
+
+def tcpdump_rate(enode):
```
Comment (saenzpa) [inline lib/topology_lib_tcpdump/library.py]: Documentation block should follow after the function definition, not before it.

File: lib/topology_lib_tcpdump/library.py [hunk 2]
```diff
@@ -49,39 +55,49 @@ def tcpdump_rate(sw):
...
+:param str interface_id: interface name.
+:param int wait_time: Time in seconds to capture with tcpdump.
+:param int num_cpu_samples: Number of CPU samples to get CPU utilization.
+ Default is 0
+"""
 
-def tcpdump_capture_interface(sw, options, interface_id, wait_time, check_cpu):
-    cmd_output = sw('ip netns exec swns tcpdump -D'.format(**locals()),
-                    'bash')
+def tcpdump_capture_interface(enode, options, interface_id, wait_time,
```
Comment (saenzpa) [inline lib/topology_lib_tcpdump/library.py]: Changing parameters in a function that has already been published will break it for anyone already using this lib. Please be sure that this change is needed, and if there's an alternative that can be backwards compatible.

File: lib/topology_lib_tcpdump/library.py [hunk 3]
```diff
@@ -49,39 +55,49 @@ def tcpdump_rate(sw):
...
     re_result = search(interface_re, cmd_output)
     assert re_result
     result = re_result.groupdict()
 
-    sw('ip netns exec swns tcpdump -ni ' + result['linux_interface'] +
-        options + ' -ttttt '
-        '> /tmp/interface.cap 2>&1 &'.format(**locals()),
-        'bash')
+    enode('ip netns exec swns tcpdump -ni ' + result['linux_interface'] +
+          options + ' -ttttt '
```
Comment (saenzpa) [inline lib/topology_lib_tcpdump/library.py]: Did this spacing pass flake8?

**Comment by saenzpa** [comment]
Hi Vivek, made some comments, please review them. Also make sure you run flake8 on the code, I think there are a couple whitespace issues.

File: lib/topology_lib_tcpdump/library.py (hunk 2 above)
Comment (vivekramamoorthy) [inline lib/topology_lib_tcpdump/library.py]: We are currently the only consumers of this API. This signature makes it more generic.

File: lib/topology_lib_tcpdump/library.py (hunk 3 above)
Comment (vivekramamoorthy) [inline lib/topology_lib_tcpdump/library.py]: yes

Comment (vivekramamoorthy) [inline lib/topology_lib_tcpdump/library.py]: Ok, i ran it on my original code, not here. Will fix them

**Comment by vivekramamoorthy** [comment]
Addressed all the comments. Let me know if this looks ok.

**Comment by saenzpa** [comment]
Great job Vivek, thanks for all the fixes.

**Pull request merged**